dataplane/unittest/shared_memory.cpp

```diff
@@ -0,0 +1,45 @@
+#include <gtest/gtest.h>
+
+#include "common/shared_memory.h"
+
+TEST(SharedMemory, UsageHugeTlb)
+{
+	bool hugetlb_enabled = common::ipc::SharedMemory::HugeTlbEnabled();
+	if (hugetlb_enabled)
+	{
+		YANET_LOG_DEBUG("HugeTLB Pages available\n");
+	}
+	else
+	{
+		YANET_LOG_DEBUG("HugeTLB Pages not available\n");
+	}
+}
+
+TEST(SharedMemory, CreateAndOpenSharedMemoryBuffer)
+{
+	std::string filename("test_shared_memory.shm");
+	uint64_t size = 1024;
+	uint8_t* buffer_writer = static_cast<uint8_t*>(common::ipc::SharedMemory::CreateBuffer(filename, size, true, 0));
+	EXPECT_TRUE(buffer_writer != nullptr);
+
+	auto [buffer_reader, size_reader] = common::ipc::SharedMemory::OpenBuffer(filename, true);
```

Comment:
Originally, `OpenBuffer` could return a pair of `nullptr, 0`, and you aren't checking it. I suggested to use `std::optional`, so this can be replaced with:
```cpp
auto opt_buffer = common::ipc::SharedMemory::OpenBuffer(filename, true);
ASSERT_TRUE(opt_buffer.has_value());
auto [buffer_reader, size_reader] = *opt_buffer;
```